.github/workflows/typespec-validation-all.yaml

```
@@ -0,0 +1,74 @@
+name: TypeSpec Validation All
+
+on:
+  push:
+    branches:
+      - main
+      - RPSaaSMaster
+      - typespec-next
+
+  pull_request:
+    branches:
+      - main
+      - RPSaaSMaster
+      - typespec-next
+    paths:
+      - .gitattributes
+      - .prettierrc.json
+      - package-lock.json
+      - package.json
+      - tsconfig.json
+      - eng/**
+      - specification/suppressions.yaml
+      - specification/common-types/**
+
+      # Workflow and workflow dependencies
+      - .github/workflows/typespec-validation-all.yml
+      - .github/actions/setup-node-ci/**
+
+  schedule:
+    # Run 4x/day
+    - cron: '0 0,6,12,18 * * * '
+
+jobs:
+  typespec-validation-all:
+    name: TypeSpec Validation All
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        folders:
+          - skip: 0
+            take: 71
+          - skip: 71
+            take: 71
+          - skip: 142
+            # Take 0 means take all
+            take: 0
```

Rather than needing to hardcode these numbers (which will constantly need to be updated), I think we can instead tell each shard the total number of shards, and the current shard number, and let the shard do the math.

```suggestion
      shards: 3
      shard: [0, 1, 2]          
```